src/app/services/bungie-api.service.ts

```diff
@@ -1,571 +1,675 @@
+                        for (let newStats of res[d.itemHash].investmentStats) {
+                            if (newStats.statTypeHash in investmentStats)
+                                investmentStats[newStats.statTypeHash] += newStats.value;
+                        }
+                    }
+                    if (r.slot != ArmorSlot.ArmorSlotClass) {
+                        const destinyItemSocketsComponents =
+                            profile.Response.itemComponents.sockets.data || {};
+                        // check if d.itemInstanceId is in destinyItemSocketsComponents
+                        if (d.itemInstanceId && d.itemInstanceId in destinyItemSocketsComponents) {
+                            const sockets: DestinyItemSocketState[] =
+                                destinyItemSocketsComponents[d.itemInstanceId || ""].sockets;
+                            var plugs = [
+                                sockets[6].plugHash,
+                                sockets[7].plugHash,
+                                sockets[8].plugHash,
+                                sockets[9].plugHash,
+                            ];
+                            r.statPlugHashes = plugs;
+                            var plm = plugs.map((k) => mods[k || ""]).filter((k) => k != null);
+                            for (let entry of plm) {
+                                for (let newStats of entry.investmentStats) {
+                                    if (newStats.statTypeHash in investmentStats)
+                                        investmentStats[newStats.statTypeHash] += newStats.value;
+                                }
+                            }
+                        } else {
+                            console.error(
+                                "Sockets data does not contain the correct item instance ID",
+                                d.itemInstanceId,
+                                profile.Response.itemComponents.sockets.data
+                            );
+                        }
+                    }
+                    r.mobility = investmentStats[2996146975];
+                    r.resilience = investmentStats[392767087];
+                    r.recovery = investmentStats[1943323491];
+                    r.discipline = investmentStats[1735777505];
+                    r.intellect = investmentStats[144602215];
+                    r.strength = investmentStats[4244567218];
+
+                    // Take a look if it really has the artifice perk
+                    if (r.perk == ArmorPerkOrSlot.SlotArtifice) {
+                        let statData = profile.Response.itemComponents.perks.data || {};
+                        let perks = (statData[d.itemInstanceId || ""] || {})["perks"] || [];
+                        const hasPerk = perks.filter((p) => p.perkHash == 229248542).length > 0;
+                        if (!hasPerk) r.perk = ArmorPerkOrSlot.None;
+                    }
+
+                    // TODO REMOVE
```

Comment:
If Remove why not removed?

Reply:
The line is required for the "trick the tool into thinking everything is an artifice piece" setting. I still need it for development. I think I will just leave it there and remove the comment.

Reply:
Or comment better ;)